BepInEx.Core/Bootstrap/LoadingSystem.cs

```diff
@@ -1,34 +1,58 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.IO;
 using System.Linq;
 using System.Reflection;
 using System.Runtime.CompilerServices;
-using System.Text;
 using System.Text.RegularExpressions;
+using BepInEx.Bootstrap;
 using BepInEx.Configuration;
 using BepInEx.Logging;
 using Mono.Cecil;
 
-namespace BepInEx.Bootstrap;
-
-public abstract class BaseChainloader<TPlugin>
+namespace BepInEx;
+
+/// <summary>
+///     The base type of any phases of a system to manage the loading of <see cref="Plugin"/> obtained from
+///     multiple <see cref="PluginProvider"/>
+/// </summary>
+/// <typeparam name="TProvider">The provider type used to obtains the <see cref="IPluginLoadContext"/> of the plugins</typeparam>
+/// <typeparam name="TPlugin">The plugin type provided by providers</typeparam>
+public abstract class LoadingSystem<TProvider, TPlugin>
+    where TProvider : PluginProvider
+    where TPlugin : Plugin
 {
-    protected static readonly string CurrentAssemblyName = Assembly.GetExecutingAssembly().GetName().Name;
-    protected static readonly Version CurrentAssemblyVersion = Assembly.GetExecutingAssembly().GetName().Version;
+    /// <summary>
+    ///     The assembly name of this loading system
+    /// </summary>
+    protected readonly string CurrentAssemblyName = Assembly.GetExecutingAssembly().GetName().Name;
+
+    /// <summary>
+    ///     The assembly version of this loading system
+    /// </summary>
+    protected readonly Version CurrentAssemblyVersion = Assembly.GetExecutingAssembly().GetName().Version;
 
-    private static Regex allowedGuidRegex { get; } = new(@"^[a-zA-Z0-9\._\-]+$");
+    /// <summary>
+    ///     The name a plugin should be referred to in the logs
+    /// </summary>
+    protected abstract string PluginLogName { get; }
+    
+    /// <summary>
+    ///     The name of the assembly load cache of this loading system
+    /// </summary>
+    protected abstract string LoadCacheName { get; }
 
-    internal BaseChainloader() { }
+    private static Regex AllowedGuidRegex => new(@"^[a-zA-Z0-9\._\-]+$");
```

Comment:
This breaks every plugin, the general convention is to do `autor.pluginname` or just `pluginname` since this is what appears on dependency errors.

Reply:
Wait, how's that possible?

All plugins and patchers were doing this on v6

Reply:
Wait, my bad, I didn't look closely and thought it was an actual GUID regex. In this case it should be fine.